app/coffee/Features/FileStore/FileStoreHandler.coffee

```diff
@@ -25,10 +25,19 @@ module.exports = FileStoreHandler =
 				timeout:fiveMinsInMs
 			writeStream = request(opts)
 			readStream.pipe writeStream
-			writeStream.on "end", callback
+
+			writeStream.on 'response', (response) ->
+				if response.statusCode not in [200, 201]
```

Comment:
Ah, I see it's hard to match on a 'not' basis. I think I'm just bikeshedding code style :), this LGTM